android/src/main/java/bbct/android/common/activity/MainActivity.java

```diff
@@ -69,18 +71,18 @@ public void onCreate(Bundle savedInstanceState) {
                     BaseballCardContract.PROJECTION, null, null, null);
 
             FragmentTransaction ft = this.getSupportFragmentManager().beginTransaction();
+            ft.add(R.id.fragment_holder, new BaseballCardList(), FragmentTags.CARD_LIST)
+                    .addToBackStack(FragmentTags.CARD_LIST)
+                    .commit();
+
             if (cursor == null || cursor.getCount() == 0) {
-                ft.add(R.id.fragment_holder, new BaseballCardDetails(), FragmentTags.EDIT_CARD);
-            } else {
-                ft.add(R.id.fragment_holder, new BaseballCardList(), FragmentTags.CARD_LIST);
-                cursor.close();
+                this.getSupportFragmentManager().beginTransaction()
+                        .replace(R.id.fragment_holder, new BaseballCardDetails(), FragmentTags.EDIT_CARD)
+                        .addToBackStack(FragmentTags.EDIT_CARD).commit();
```

Comment:
Let me try when I get home.

Reply:
Commiting these transaction at the end of the if block would cause the Navigate up button to be disabled, I've tried debugging and I could only see that ```getSupportFragmentManager().getBackStackEntryCount() = 1```
What happens under the hood is that the BaseballCardlist fragment would never get rendered properly resulting into the menu buttons not showing and bug when navigating up, as opposed having two transactions where we will get two backstack entry count and each of the fragments are rendered which results in what we desire to have UX wise. I've just updated the code now instead of having ft we will keep reference to fragmentmanager.